### ISSUE TITLE
What is a suitable value for the loss when running color_syncnet_train.py training? 

### ISSUE THREAD

**Issue by bobo-wmdigit** [Rudrabha/Wav2Lip#484]
My loss is still fluctuating up and down.
![image](https://user-images.githubusercontent.com/80187914/230534552-82241d6e-26d1-43ad-9431-2b6a7cf56ccb.png)

**Comment by fttz** [comment]
hparams.py set number of epoch as follow:
`nepochs=200000000000000000,  ### ctrl + c, stop whenever eval loss is consistently greater than train loss for ~10 epochs`
Btw, what is the minimum loss when you train syncnet？

**Comment by YinonDouchanClarity** [comment]
SyncNet uses binary cross entropy on the cosine similarity between the output video and audio embeddings. More formally, given ground truth y (0 for out of sync audio-video pair, 1 for in-sync) and s, the similarity score (between 0 and 1, 0 for dissimilar embeddings, 1 for similar), the loss is equal to L = -y * ln(s) - (1 - y) * ln(1 - s). 

To answer your questions:

What's the minimum loss? The minimum loss is when s=1 when y=1 and s=0 when y=0 which will be 0 and the maximum is unbounded.

What's a suitable value for the loss? If your model is no better than random guess, you'd expect s to be 0.5 on average, which will make L around -ln(0.5) ~ 0.6931. This is why many people posted issues stating that their loss can't pass 0.69. Therefore, any evaluation loss which is lower than 0.69 indicates that your model performs better than random guess. The authors state in the README that when training the lip-sync generator (after training syncnet), the sync loss (same BCE loss coming from SyncNet) should be lower than 0.2. Therefore, I'd recommend that when training syncnet, your eval loss should be lower than that.

**Comment by zhanglonghao1992** [comment]
@YinonDouchanClarity  Have you trained on other dataset and got a eval loss around 0.2?
